README.md

```diff
@@ -1,2 +1,110 @@
 # api.webmaker.org
-Services for Webmaker
+
+This is where we'll build the service for saving, publishing, and searching projects created by the Webmaker Android application.
+
+## Authentication Requirements
+
+### Basic requirements
+
+- should use the `login.webmaker.org` Oauth2 provider
+- must work on Android and Android web views (client side/mobile token-based strategy)
+- if possible, a dev instance of Oauth2 provider should be accessible without specific origin requirements or access to privileged credentials
+
+### User Roles
+
+#### Unauthenticated users
+- should be able to try the editor, initiate remixes
+- should have read-only access to search/discovery features
+- should have read-only access to view published `Project`s
+- should be able to initiate a "create account sequence"
+
+#### Authenticated users
+- should be able to create and save new `Project`s
+- should be able to read, update, and delete all own `Project`s
+
+#### Admins
+- should be able to delete `Project`s by other users
+- should be able to set "banned/removed" or "featured" status on `Project`s
+
+## Data Model
+- basic unit is a `Project`
+- `Project`s:
+  - are associated with a user (the creator)
+  - contain **metadata** fields such as date created, date updated, title, description, etc.
+  - contain **content**, which is a list of "blocks", each containing metadata
+  - have a **version**, indicating whether metadata and the metadata of each block are compatible with the current version of the app, which is auto-generated and read-only
+  - have **privileged metadata**, such as featured or banned status, that is only editable by admin level accounts
+- if possible, `Project`s should be able to be migrated based on their version numbers.
+- a more specific definition of this data model needs to be determined once design is finalized.
+
+## API Requirements
+
+### Projects API
+
+Basic needs: retrieve and edit projects belonging to a user
+Authentication: unauthed: no acess; authed: read-write; admin: read-write
+
+### Create/Read/Update/Delete projects
+- create a new `Project`, return an ID
+- read/update/delete a `Project` by ID
+- should implement "soft delete"
```

Comment:
hm. we need to have a way to undo deletions, but there are other ways to handle this. happy to leave it up for discussion